Code/GraphMol/Substruct/test1.cpp

```diff
@@ -1589,8 +1589,212 @@ void testGithubIssue1489() {
   BOOST_LOG(rdErrorLog) << "  done" << std::endl;
 }
 
+void testGithub2570() {
+  BOOST_LOG(rdInfoLog) << "-------------------------------------" << std::endl;
+  BOOST_LOG(rdInfoLog)
+      << "Testing Github #2570: issue pointed out by Greg in the PR."
+      << std::endl;
+
+  bool uniquify = true;
+  bool recursionPossible = true;
+  bool useChirality = true;
+  {
+    const auto mol = R"(C[C@](Cl)(Br)F)"_smiles;
+
+    {
+      const auto query = R"([C@](Cl)(Br)F)"_smarts;
+      std::vector<MatchVectType> matches;
+      TEST_ASSERT(SubstructMatch(*mol, *query, matches, uniquify,
+                                 recursionPossible, useChirality));
+    }
+    {
+      const auto query = R"([C@@](Cl)(Br)F)"_smarts;
+      std::vector<MatchVectType> matches;
+      TEST_ASSERT(!SubstructMatch(*mol, *query, matches, uniquify,
+                                  recursionPossible, useChirality));
+    }
+    {  // Swap order of a pair of atoms
+      const auto query = R"([C@](Br)(Cl)F)"_smarts;
+      std::vector<MatchVectType> matches;
+      TEST_ASSERT(!SubstructMatch(*mol, *query, matches, uniquify,
+                                  recursionPossible, useChirality));
+    }
+    {
+      const auto query = R"([C@@](Br)(Cl)F)"_smarts;
+      std::vector<MatchVectType> matches;
+      TEST_ASSERT(SubstructMatch(*mol, *query, matches, uniquify,
+                                 recursionPossible, useChirality));
+    }
+    {  // Try smaller fragments
+      const auto query = R"([C@](Cl)Br)"_smarts;
+      std::vector<MatchVectType> matches;
+      TEST_ASSERT(SubstructMatch(*mol, *query, matches, uniquify,
+                                 recursionPossible, useChirality));
+    }
+    {
+      const auto query = R"([C@@](Cl)Br)"_smarts;
+      std::vector<MatchVectType> matches;
+      TEST_ASSERT(!SubstructMatch(*mol, *query, matches, uniquify,
```

Comment:
I think that the way the logic of tetrahedral chirality goes, matching less than three neighbors of a 4-coordinate chiral atom is *always* a match. There just isn't sufficient information in the query to tell you otherwise.

Reply:
Yeah, that makes sense, too. I think I took the order of the SMARTS string too literally here, thinking of matching a literal CW chiral atom in which the first two consecutive bonds to neighbors are Cl and Br, or the opposite: a CCW atom with the bonds in reverse order... anyway, as I said, I'm not 100% sure about the 'fragments' matching making sense.